extensions/amp-auto-ads/0.1/test/test-ad-strategy.js

```diff
@@ -0,0 +1,207 @@
+        expect(placements.length).to.equal(2);
+
+        const adStrategy = new AdStrategy('ad-network-type', placements, [
+          {
+            name: 'custom-att-1',
+            value: 'val-1',
+          },
+          {
+            name: 'custom-att-2',
+            value: 'val-2',
+          },
+        ]);
+
+        adStrategy.run().then(success => {
+          expect(success).to.equal(true);
+          expect(anchor1.childNodes.length).to.equal(1);
+          expect(anchor2.childNodes.length).to.equal(0);
+          const adElement = anchor1.childNodes[0];
+          expect(adElement.tagName).to.equal('AMP-AD');
+          expect(adElement.getAttribute('type')).to.equal('ad-network-type');
+          expect(adElement.getAttribute('data-custom-att-1')).to.equal('val-1');
+          expect(adElement.getAttribute('data-custom-att-2')).to.equal('val-2');
+          done();
+        });
+      });
+
+  it('should place the second ad when placing the first one fails', done => {
+    const anchor1 = document.createElement('div');
+    anchor1.id = 'anId1';
+    container.appendChild(anchor1);
+
+    const anchor2 = document.createElement('div');
+    anchor2.id = 'anId2';
+    container.appendChild(anchor2);
+
+    const configObj = {
+      placements: [
+        {
+          anchor: {
+            selector: 'DIV#anId1',
+          },
+          pos: 2,
+          type: 1,
+        },
+        {
+          anchor: {
+            selector: 'DIV#anId2',
+          },
+          pos: 2,
+          type: 1,
+        },
+      ],
+    };
+    const placements = getPlacementsFromConfigObj(window, configObj);
+
+    expect(placements.length).to.equal(2);
+    sandbox.stub(placements[0], 'placeAd', () => {
+      return Promise.resolve(PlacementState.REIZE_FAILED);
+    });
+
+    const adStrategy = new AdStrategy('ad-network-type', placements, [
+      {
+        name: 'custom-att-1',
+        value: 'val-1',
+      },
+      {
+        name: 'custom-att-2',
+        value: 'val-2',
+      },
+    ]);
+
+    adStrategy.run().then(success => {
+      expect(success).to.equal(true);
+      expect(anchor1.childNodes.length).to.equal(0);
+      expect(anchor2.childNodes.length).to.equal(1);
+      const adElement = anchor2.childNodes[0];
+      expect(adElement.tagName).to.equal('AMP-AD');
+      expect(adElement.getAttribute('type')).to.equal('ad-network-type');
+      expect(adElement.getAttribute('data-custom-att-1')).to.equal('val-1');
+      expect(adElement.getAttribute('data-custom-att-2')).to.equal('val-2');
+      done();
+    });
+  });
+
+  it('should report strategy as unsuccessful when unable to place either ad',
+      done => {
+        const anchor1 = document.createElement('div');
+        anchor1.id = 'anId1';
+        container.appendChild(anchor1);
+
+        const anchor2 = document.createElement('div');
+        anchor2.id = 'anId2';
+        container.appendChild(anchor2);
```

Comment:
I started doing this, but I feel like it got more messy and less readable. I need to have a reference to anchor1 and anchor2 to do the assertions below. If I specify the DOM structure using innerHTML, then I have to do:
const anchor1 = document.getElementById('anId1');
const anchor2 = document.getElementById('anId2');
at some point, which seems a little clunky and means more lines have to be looked at to figure out what's what in the test case.
I've changed the ID names to be 'anchorId1' and 'anchorId2', as this hopefully makes it more readable.